File: app/src/main/java/com/uramonk/androidtemplateapp/ModuleInjector.java
package com.uramonk.androidtemplateapp;

import com.uramonk.androidtemplateapp.component.DaggerWeatherComponent;
import com.uramonk.androidtemplateapp.component.WeatherComponent;
import com.uramonk.androidtemplateapp.module.NetworkModule;
import com.uramonk.androidtemplateapp.module.WeatherModule;

/**
 * Created by uramonk on 2016/06/22.
 */
public class ModuleInjector {
    /**
     * Components
     */
    private WeatherComponent weatherComponent;

    public static ModuleInjector instance = new ModuleInjector();

    private ModuleInjector() {

    }

    public static ModuleInjector getInstance() {
        return instance;
    }

    public WeatherComponent getWeatherComponent() {
        if (weatherComponent == null) {
            weatherComponent = DaggerWeatherComponent.builder()
                    .networkModule(new NetworkModule(Constants.OPEN_WEATHER_MAP_URL))
                    .weatherModule(new WeatherModule())
                    .build();
        }
        return weatherComponent;
    }
}

File: app/src/main/java/com/uramonk/androidtemplateapp/ModuleInjector.kt
package com.uramonk.androidtemplateapp

import com.uramonk.androidtemplateapp.di.component.DaggerWeatherComponent
import com.uramonk.androidtemplateapp.di.component.WeatherComponent
import com.uramonk.androidtemplateapp.di.module.NetworkModule
import com.uramonk.androidtemplateapp.di.module.WeatherModule

/**
 * Created by uramonk on 2016/06/22.
 */

class ModuleInjector private constructor() {
    /**
     * Singleton Components
     */
    object SingletonWeatherComponent {
        val instance: WeatherComponent = DaggerWeatherComponent.builder()
                .networkModule(NetworkModule(Constants.OPEN_WEATHER_MAP_URL))
                .weatherModule(WeatherModule())
                .build()
    }
}
